jellyfin_kodi/jellyfin/ws_client.py

```diff
@@ -54,24 +55,44 @@ def run(self):
         LOG.info("Websocket url: %s", wsc_url)
 
         self.wsc = websocket.WebSocketApp(wsc_url,
+                                          on_open=lambda ws: self.on_open(ws),
                                           on_message=lambda ws, message: self.on_message(ws, message),
                                           on_error=lambda ws, error: self.on_error(ws, error))
-        self.wsc.on_open = self.on_open
 
+        retry_count = 0
         while not self.stop:
 
+            time.sleep(retry_count * 5)
```

Comment:
If I read this correctly, the connection time will increase by 5s from 0-60s on every connection (attempt), which means that long-running clients will always wait 60s before connecting.

`retry_count` should probably be reset either in `on_connect` or `on_message`.

I assume `ping_interval` means the client is sending data down the TCP pipe every 10s? That's good, because in my experience, TCP sockets have a tendency to hang forever on read if no data is attempted sent down a socket that is gone on the other end.

Reply:
Only if the connection has failed over and over.  The first attempt, it doesn't wait at all.  The second attempt, 5 seconds, third attempt 10 seconds, etc etc etc.  So the only time it's waiting for a minute before trying again is if the connection has failed at least 12 times previously.  My thoughts were if the connection just dropped, initialize it again as soon as possible, but if the server is restarting or down for maintenance it doesn't do any good to just keep hammering every few seconds to get back in.  And obviously the count will reset whenever Kodi is restarted.

Reply:
Yes, I understood the intention, but what I am saying is that it looks like the counter is never reset, so that on any subsequent reconnects it waits *at least* 5s, as that's in an infinite loop.

It makes sense to reset the counter on a successful connection, so that the first reconnect doesn't take 5s, and the second 10s if that isn't needed.

If the Kodi client runs for a week, but the connection to the server dies briefly twice a day, you currently have a guaranteed 60s delay before it attempts to reconnect again (after whatever timeout happened at the TCP/websocket layers).

I suggest moving the retry_count to the class instance, and setting it to 0 (or -1, but make sure to skip the sleep if <0 then) after successfully connecting.